### PR TITLE
New network Top level resource: VpnServerConfiguration related APIs

### PR DESCRIPTION
**Pull request by Nilambari**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by Nilambari** [comment]
Label: DoNotMerge. to add. I have opened this PR for ARM team to take a look and approve ARM manifest update for canary region for testing. We do not want to merge this PR yet.

**Comment by Nilambari** [comment]
New PR is opened to master as requested. Previous PR:- https://github.com/Azure/azure-rest-api-specs/pull/6616 shows review comments given. @KrisBash Can you Please take a look at this PR? Thanks a lot!

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please model certificates as properties (not embedded child resources) and ensure that the root resource (vpnServerConfigurations) has mandatory properties - id, name, type, tags, properties, location

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnServerConfigurationGet.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			  }
+			],
+			"vpnServerConfigVpnClientRevokedCertificates": [
+			  {
+				  "name": "vpnServerConfigVpnClientRevokedCert1",								  
+				  "Thumbprint": "83FFBFC8848B5A5836C94D0112367E16148A286F"
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnServerConfigurationGet.json]: Thumbprint -> thumbprint.
Bunch of other hits for the same in the  other examples

Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/examples/VpnServerConfigurationGet.json]: Just fixed the same.

**Comment by Nilambari** [comment]
> Please model certificates as properties (not embedded child resources) and ensure that the root resource (vpnServerConfigurations) has mandatory properties - id, name, type, tags, properties, location

Incorporated review comments as per discussion. Build is passing now. @KrisBash Can you take a look?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json [hunk 2]
```diff
@@ -810,6 +810,248 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnServerConfigurations/{vpnServerConfigurationName}": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: What happened to the PATCH api? That is needed, at least for `tags` but ideally for all mutable properties

**Review by KrisBash — CHANGES_REQUESTED** [review]
Thanks for the updates. I think this is good once the missing PATCH tags api is added back in

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json (hunk 2 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: We don't need PATCH api the scenario to update tags is not valid for this resource.  Also, we do update mutable properties using PUT api only, this is similar to other resource types's apis design.

Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json]: Added PATCH api back and build is passing now. Thanks!

**Review by KrisBash — APPROVED** [review]
lgtm - thanks for the updates

**Comment by sergey-shandar** [comment]
@Nilambari are still working on this PR?

**Comment by sergey-shandar** [comment]
@KrisBash could you remove `DoNotMerge` label if it's not applicable anymore?

**Comment by sergey-shandar** [comment]
@KrisBash could you remove `DoNotMerge`?

**Comment by sergey-shandar** [comment]
@Nilambari what's the status of the PR, do you plan to close or merge this PR?

**Comment by number213** [comment]
Please also retarget to `network-july-release`

**Comment by sergey-shandar** [comment]
@Nilambari I'm closing this PR. Please, open a new one if you still need it and you may refer to this PR to simplify review process.

**Pull request closed without merging**